test/fork/merkle-lockup/MerkleLockupLD.t.sol

```diff
@@ -0,0 +1,232 @@
+        LockupDynamic.Stream expectedStream;
+        uint256 expectedStreamId;
+        uint256[] indexes;
+        uint256 leafPos;
+        uint256 leafToClaim;
+        ISablierV2MerkleLockupLD merkleLockupLD;
+        bytes32 merkleRoot;
+        address[] recipients;
+        uint256 recipientsCount;
+        uint40[] streamDurations;
+        LockupDynamic.SegmentWithDuration[][] segments;
+        LockupDynamic.Segment[][] segmentsWithTimestamps;
+    }
+
+    // We need the leaves as a storage variable so that we can use OpenZeppelin's {Arrays.findUpperBound}.
+    uint256[] public leaves;
+
+    function testForkFuzz_MerkleLockupLD(Params memory params) external {
+        vm.assume(params.admin != address(0) && params.admin != users.admin);
+        vm.assume(params.expiration == 0 || params.expiration > block.timestamp);
+        vm.assume(params.leafData.length > 1);
+        params.posBeforeSort = _bound(params.posBeforeSort, 0, params.leafData.length - 1);
+        assumeNoBlacklisted({ token: address(asset), addr: params.admin });
+
+        /*//////////////////////////////////////////////////////////////////////////
+                                          CREATE
+        //////////////////////////////////////////////////////////////////////////*/
+
+        Vars memory vars;
+
+        vars.recipientsCount = params.leafData.length;
+        vars.amounts = new uint128[](vars.recipientsCount);
+        vars.indexes = new uint256[](vars.recipientsCount);
+        vars.recipients = new address[](vars.recipientsCount);
+        vars.segments = new LockupDynamic.SegmentWithDuration[][](vars.recipientsCount);
+        vars.segmentsWithTimestamps = new LockupDynamic.Segment[][](vars.recipientsCount);
+        vars.streamDurations = new uint40[](vars.recipientsCount);
+
+        for (uint256 i = 0; i < vars.recipientsCount; ++i) {
+            vm.assume(
+                params.leafData[i].segments.length > 0
+                    && params.leafData[i].segments.length < defaults.MAX_SEGMENT_COUNT()
+            );
+            vars.indexes[i] = params.leafData[i].index;
+
+            // Bound each leaf's segment duration to avoid overflows.
+            fuzzSegmentDurations(params.leafData[i].segments);
+
+            uint256 segmentCount = params.leafData[i].segments.length;
+            for (uint256 j = 0; j < segmentCount; ++j) {
+                // Bound each leaf's segment amount so that `aggregateAmount` does not overflow.
+                params.leafData[i].segments[j].amount = uint128(
+                    _bound(
+                        params.leafData[i].segments[j].amount,
+                        1,
+                        MAX_UINT128 / (vars.recipientsCount * segmentCount) - 1
+                    )
+                );
+
+                vars.amounts[i] += params.leafData[i].segments[j].amount;
+                vars.streamDurations[i] += params.leafData[i].segments[j].duration;
```

Comment:
Right. Changed it to the following:

```solidity
vars.streamDurations[i] = vars.segmentsWithTimestamps[i][params.leafData[i].segments.length - 1].timestamp - getBlockTimestamp();
```

Reply:
we don't need `streamDurations` at all, we can remove it from vars struct

you declared this only for here:

https://github.com/sablier-labs/v2-periphery/blob/dd1f751750cf2f312afc3ff9bddefc1ab666e51f/test/fork/merkle-lockup/MerkleLockupLD.t.sol#L183

but we can use instead: 
```solidity
endTime: vars.segmentsWithTimestamps[params.posBeforeSort][params.leafData[i].segments.length - 1].timestamp
```

Reply:
I understand what you are saying now. I realized I could also have declared it as `uint40 streamDurations` instead of `uint40[] streamDurations` since we only need value at a specific index. 

If we do not use variable, then this is how it would look with auto formatting, Are you OK with this? 

<img width="952" alt="Screenshot 2024-02-11 at 23 33 57" src="https://github.com/sablier-labs/v2-periphery/assets/6676622/cbfeec50-469b-4e39-a4fa-53c6ddd754ea">

Reply:
> If we do not use variable, then this is how it would look with auto formatting, Are you OK with this?

i am not sure i understand this message


>  I could also have declared it as uint40 streamDurations instead of uint40[] streamDurations since we only need value at a specific index.

we can add a variable (not the array) in the `Vars`  struct `endTime` to have a nicer formatting

Reply:
> i am not sure i understand this message

I was talking about formatting.

> we can add a variable (not the array) in the Vars struct endTime to have a nicer formatting

Agree.